### PR TITLE
Koji NVR endpoint

### PR DESCRIPTION
**Pull request by MartinBasti**
TODO:

- [x] unittests
- [x] README

Allows to download extracted manifests archive from koji (build by OSBS)
and uploads manifest artifact to quay.io

* OSBS-6687

Signed-off-by: Martin Bašti <mbasti@redhat.com>

### CODE REVIEW

File: omps/api/v1/push.py [hunk 1]
```diff
@@ -149,21 +186,41 @@ def push_zipfile(organization, repo, version=None):
...
+        organization=organization,
+        repo=repo,
+        version=version,
+        extract_manifest_func=partial(extract_zip_file_from_request, request)
+    )
+
+
+@API.route("/<organization>/<repo>/koji/<nvr>", defaults={"version": None},
+           methods=('POST',))
+@API.route("/<organization>/<repo>/koji/<nvr>/<version>", methods=('POST',))
```
Comment (csomh) [inline omps/api/v1/push.py]: Wondering if a route like `/<organization>/<repo>/<version>/koji/<nvr>` would work and would make more sense...

File: omps/errors.py [hunk 2]
```diff
@@ -49,6 +49,26 @@ class OMPSAuthorizationHeaderRequired(OMPSError):
     code = 403
 
 
+class KojiNVRBuildNotFound(OMPSError):
+    """Requested build not found in koji"""
+    code = 404
```
Comment (csomh) [inline omps/errors.py]: I think this would rather be a 400 Bad Request

File: omps/errors.py [hunk 3]
```diff
@@ -49,6 +49,26 @@ class OMPSAuthorizationHeaderRequired(OMPSError):
...
+
+
+class KojiNotAnOperatorImage(OMPSError):
+    """Requested build is not an operator image"""
+    code = 500
```
Comment (csomh) [inline omps/errors.py]: I think this would rather be a 400 Bad Request

File: omps/api/v1/push.py (hunk 1 above)
Comment (MartinBasti) [reply omps/api/v1/push.py]: I tried to keep the same patter as for zipfile route

File: omps/errors.py (hunk 3 above)
Comment (MartinBasti) [reply omps/errors.py]: ok

File: omps/errors.py (hunk 2 above)
Comment (MartinBasti) [reply omps/errors.py]: ok

Comment (MartinBasti) [reply omps/errors.py]: I changed mind with this one, 404 is better IMO

Comment (csomh) [reply omps/errors.py]: ok :)

File: omps/api/v1/push.py (hunk 1 above)
Comment (csomh) [reply omps/api/v1/push.py]: I see.

The reason I asked is b/c in this layout koji and quay elements are mixed:

```
/<organization>/<repo>/koji/<nvr>/<version>
/^---quay-------------/^koji-----/^--quay--
```

But to make it consistent probably would require to change the zipfile route, too. And we don't want this atm.

So let's leave it this way for now :)

File: .travis.yml [hunk 4]
```diff
@@ -13,5 +13,7 @@ matrix:
     - dist: xenial
       python: "3.7"
       env: TOXENV=flake8
+before_install:
+  - sudo apt-get install -y rpm  # for rpm-py-installer (koji requirement)
```
Comment (csomh) [inline .travis.yml]: Nice!

On my fedora box, in order to make `tox` work, had to `dnf install popt-devel`. Maybe would make sense to mention this in the README.

File: omps/api/v1/push.py [hunk 5]
```diff
@@ -35,7 +38,56 @@ def validate_allowed_extension(filename):
...
 
-def extract_zip_file(
+def _extract_zip_file(
+    filepath, target_dir,
+    max_uncompressed_size=DEFAULT_ZIPFILE_MAX_UNCOMPRESSED_SIZE
+):
+    """Extract zip file
+
+    :param filepath: zip archive file
+    :param target_dir:
```
Comment (csomh) [inline omps/api/v1/push.py]: docstrings are incomplete.

File: omps/constants.py [hunk 6]
```diff
@@ -13,3 +13,5 @@
 DEFAULT_RELEASE_VERSION = '1.0.0'
 
 ALLOWED_EXTENSIONS = {".zip", }
+
+KOJI_OPERATOR_MANIFESTS_ARCHIVE_KEY = 'operator_manifests_archive'
```
Comment (csomh) [inline omps/constants.py]: Is this string hardcoded in OSBS or it's a configuration value. If the later, than might be better to make it a configuration value for OMPS, too.

Comment (MartinBasti) [reply omps/constants.py]: It is hardcoded

File: .travis.yml (hunk 4 above)
Comment (MartinBasti) [reply .travis.yml]: Is it missing tox dependency? Or which package needs that? I had no issues with installing tox

Comment (csomh) [reply .travis.yml]: Ok, so the right dependency on fedora is `rpm-devel`.

Without this, on the system, tox will fail to `pip install` OMPS in the test environments, as the installation of rpm-py-installer (a dependency of koji) fails.

File: omps/constants.py (hunk 6 above)
Comment (csomh) [reply omps/constants.py]: Then no change needed here.

File: omps/api/v1/push.py (hunk 5 above)
Comment (MartinBasti) [reply omps/api/v1/push.py]: fixed

**Comment by MartinBasti** [comment]
min coverage limit set to 90% :green_heart:

File: .travis.yml (hunk 4 above)
Comment (MartinBasti) [reply .travis.yml]: dev docs updated

**Pull request merged**